server/aesdsocket.c

```diff
@@ -0,0 +1,376 @@
+            		freeaddrinfo(server_addr_info);
+        	}
+        	close_n_exit(EXIT_FAILURE);
+    	}
+    	syslog(LOG_INFO, "Set port reuse option");
+	
+	// Bind socket and port
+    	if (bind(sock_fd, server_addr_info->ai_addr, server_addr_info->ai_addrlen) != 0)
+    	{
+        	syslog(LOG_PERROR, "ERROR: Failed to bind");
+        	if (server_addr_info != NULL)
+        	{
+            		freeaddrinfo(server_addr_info);
+        	}
+        	close_n_exit(EXIT_FAILURE);
+    	}
+    	syslog(LOG_INFO, "Bind Successful");
+	
+	// Free server_addr_info after bind
+    	if (server_addr_info != NULL)
+    	{
+        	freeaddrinfo(server_addr_info);
+        	syslog(LOG_INFO, "Memory Free");
+    	}
+	
+	// After acquiring address and binding depending on -d param
+	if (run_as_daemon) 
+	{
+		syslog(LOG_INFO, "Running as daemon");
+        	run_as_daemon_func();
+    	}
+	
+	// Listen on Socket
+	if (listen(sock_fd, 10) == -1) 
+	{
+        	syslog(LOG_ERR, "ERROR: Failed to listen");
+        	close_n_exit(EXIT_FAILURE);
+    	}
+	
+	
+	char buffer[BUFF_SIZE] = {'\0'};
+    	bool packet_complete = false;
+    	int recv_bytes = 0;
+    	int written_bytes = 0;
+    	
+	while(!exit_flag)
+	{
+        	struct sockaddr_in client_addr;
+        	socklen_t client_addr_len = sizeof(client_addr);
+        	accepted_fd = accept(sock_fd, (struct sockaddr*)&client_addr, &client_addr_len);
+        	if (accepted_fd == -1) 
+        	{
+            		syslog(LOG_WARNING, "WARNING: Failed to accept, retrying ...");
+            	continue; // Continue accepting connections
+        	}
+        	syslog(LOG_INFO, "connection accepted: %d",accepted_fd);
+        	
+        	// Log accepted connection
+        	// converts binary ip address to string format
+        	char client_ip[INET_ADDRSTRLEN];
+        	if (inet_ntop(AF_INET, &(client_addr.sin_addr), client_ip, INET_ADDRSTRLEN) == NULL)
+        	{
+        		syslog(LOG_ERR, "ERROR: Failed to get ip");	
+        	}
+        	syslog(LOG_INFO, "Accepted connection from %s", client_ip);
+        	
+        	// Open file for aesdsocketdata
+    		file_fd = open(FILENAME, O_CREAT | O_RDWR | O_APPEND, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+
+    		if (file_fd == -1)
+    		{
+        		syslog(LOG_ERR, "ERROR: Failed to create/open file");
+        		close_n_exit(EXIT_FAILURE);
+    		}
+    		syslog(LOG_INFO, "File open successfully :%d",file_fd);
+        	
+        	do
+        	{
+        		memset(buffer, 0, BUFF_SIZE);
+            		// recieve data from client
+            		recv_bytes = recv(accepted_fd, buffer, BUFF_SIZE, 0);
+            		if (recv_bytes == -1)
+            		{
+                		syslog(LOG_ERR, "ERROR: Failed to recieve byte from client");
+        			close_n_exit(EXIT_FAILURE);
+            		}
+            		syslog(LOG_INFO, "data recieved");
+            		
+            		// write the string received to the file 
+            		written_bytes = write(file_fd, buffer, recv_bytes);
```

Comment:
@JayashCUB with this design you will write to the file before the packet is complete, which may cause problems with future assignments.